src/base/bittorrent/session.cpp

```diff
@@ -4513,6 +4514,43 @@ void Session::startUpTorrents()
         if (needStore)
              m_resumeDataStorage->store(torrentID, resumeData);
 
+        const QString category = resumeData.category;
+        bool isCategoryRecovered = recoveredCategories.contains(category);
+        if (!category.isEmpty() && (isCategoryRecovered || !m_categories.contains(category)))
+        {
+            if (!isCategoryRecovered)
+            {
+                if (addCategory(category))
+                {
+                    recoveredCategories.insert(category);
+                    isCategoryRecovered = true;
+                    LogMsg(tr("Inconsistent data is detected. Category '%2' is assigned to some torrent(s) but doesn't exist. It will be recovered with default settings.")
```

Comment:
>It will be recovered with default settings

Maybe `It will be reset to default category`?

Reply:
> > It will be recovered with default settings
> 
> Maybe `It will be reset to default category`?

This is not what is meant. If the Foo category is not found, then It will be recovered with default settings (i.e. created again having default settings). We will reset torrents category to default one only if category cannot be created.

Reply:
Then perhaps `Its settings will be reset to default`?